src/Collapse/include/gudhi/Flag_complex_sparse_matrix.h

```diff
@@ -0,0 +1,376 @@
+/*    This file is part of the Gudhi Library - https://gudhi.inria.fr/ - which is released under MIT.
+ *    See file LICENSE or go to https://gudhi.inria.fr/licensing/ for full license details.
+ *    Author(s):       Siddharth Pritam
+ *
+ *    Copyright (C) 2020 Inria
+ *
+ *    Modification(s):
+ *      - 2020/03 Vincent Rouvreau: integration to the gudhi library
+ *      - YYYY/MM Author: Description of the modification
+ */
+
+#ifndef FLAG_COMPLEX_SPARSE_MATRIX_H_
+#define FLAG_COMPLEX_SPARSE_MATRIX_H_
+
+#include <gudhi/graph_simplicial_complex.h>
+#include <gudhi/Debug_utils.h>
+
+#include <boost/functional/hash.hpp>
+#include <boost/graph/adjacency_list.hpp>
+
+#include <Eigen/Sparse>
+
+#ifdef GUDHI_USE_TBB
+#include <tbb/parallel_sort.h>
+#endif
+
+#include <iostream>
+#include <utility>  // for std::pair
+#include <vector>
+#include <unordered_map>
+#include <unordered_set>
+#include <set>
+#include <tuple>  // for std::tie
+#include <algorithm>  // for std::includes
+#include <iterator>  // for std::inserter
+
+namespace Gudhi {
+
+namespace collapse {
+
+/**
+ * \class Flag_complex_sparse_matrix
+ * \brief Flag complex sparse matrix data structure.
+ *
+ * \ingroup collapse
+ *
+ * \details
+ * This class stores a <a target="_blank" href="https://en.wikipedia.org/wiki/Clique_complex">Flag complex</a>
+ * in an <a target="_blank" href="https://eigen.tuxfamily.org/dox/group__TutorialSparse.html">Eigen sparse matrix</a>.
+ *
+ * \tparam Vertex type must be a signed integer type. It admits a total order <.
+ * \tparam Filtration type for the value of the filtration function. Must be comparable with <.
+ */
+template<typename Vertex, typename Filtration>
+class Flag_complex_sparse_matrix {
```

Comment:
I am not fond of the name of the class (and file). I know it is already in namespace `collapse`, but this isn't a datastructure to store flag complexes, this is a utility to collapse edges. To try and keep part of the current name, maybe Flag_complex_edge_collapser? Feel free to suggest other names.

Reply:
You are right. I also renamed variable names when constructing a Flag_complex_edge_collapser on fcd06dde